server/src/main/java/io/crate/execution/ddl/tables/TransportCloseTable.java

```diff
@@ -164,15 +165,31 @@ static ClusterState closeRoutingTable(ClusterState currentState,
             updatedState = ClusterState.builder(currentState).metadata(metadata).build();
         }
 
+        RelationMetadata.Table table = updatedState.metadata().getRelation(target.table());
         List<String> partitionValues = target.partitionValues();
+        final Metadata.Builder metadata;
         if (partitionValues.isEmpty()) {
             updatedState = ddlClusterStateService.onCloseTable(updatedState, target.table());
+            metadata = Metadata.builder(updatedState.metadata());
+            metadata.addTable(
```

Comment:
Seems like I don't understand the pattern here, wouldn't this add a table multiple times if I close it multiple times? Isn't `Metadata.builder(updatedState.metadata());` adding existing tables to the schema as well?  (same questions relates to all other action changes)

Reply:
`.addTable` overrides tables with the same name, so this is only changing the state to closed as it re-uses all other settings from the table.

I could rename `.addTable` to `.setTable` to clarify that?

Reply:
`addOrReplace` seems overly verbose to me. I'd rather go with existing terms for this behavior which would be either `set` (e.g. used the ObjectType builder) or `put` (used for maps)

Reply:
Also ok for me. But would be great if we can clarify this for `addToSchema` as well.